src/prometheus_server.py

```diff
@@ -35,7 +46,7 @@ def reload_configuration(self) -> bool:
         """
         url = f"{self.base_url}/-/reload"
         try:
-            response = post(url, timeout=self.api_timeout)
+            response = post(url, timeout=self.api_timeout, allow_redirects=False)
```

Comment:
Why not allow redirects? Might be worth a comment.

Reply:
It was actually not needed anymore. The reason why at some point in making the PR I thought I needed it, is that under some circumstances (which we solved with `urljoin`), the URL would not be well-formed and Prometheus would give me a 301 back, which then resulted in a 405, as the redirect would use a `GET` method that is not supported by the reload API.